File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        std::vector<int>::iterator low,up;
        low = lower_bound(nums.begin() , nums.end() , target);
        up = upper_bound(nums.begin() , nums.end() , target);
        int first = low - nums.begin();
        int last = up - nums.begin()-1;
        if(low == up && (first >= nums.size() || nums[first] != target)) return {-1 , -1};
       

        return {first , last};
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if(len(nums) == 0): return [-1 , -1]
        a = bisect_left(nums , target)
        b = bisect_right(nums , target)
        if(a == b and ( a >= len(nums) or nums[a] != target)) : return [-1 , -1]
        return [a , b - 1]